cli/command/task_test.go

```diff
@@ -57,6 +57,54 @@ func TestCreateTask(t *testing.T) {
 	}
 }
 
+func TestCreateTask_stateful(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	command := NewTaskCommand(base.CommandBase()).Command()
+
+	base.Resolver.EXPECT().
+		Resolve("environment", "env_name").
+		Return([]string{"env_id"}, nil)
+
+	base.Resolver.EXPECT().
+		Resolve("deploy", "dpl_name").
+		Return([]string{"dpl_id"}, nil)
+
+	overrides := []models.ContainerOverride{
+		{ContainerName: "c1", EnvironmentOverrides: map[string]string{"k1": "v1"}},
+		{ContainerName: "c2", EnvironmentOverrides: map[string]string{"k2": "v2"}},
+	}
+
+	validateRequest := func(req models.CreateTaskRequest) {
+		assert.Equal(t, "tsk_name", req.TaskName)
+		assert.Equal(t, "env_id", req.EnvironmentID)
+		assert.Equal(t, "dpl_id", req.DeployID)
+
+		assert.Len(t, req.ContainerOverrides, len(overrides))
```

Comment:
pretty sure you can just use [ElementsMatch](https://godoc.org/github.com/stretchr/testify/assert#ElementsMatch) `assert.ElementsMatch(t, req.ContainerOverrides, overrides)`

Reply:
Oh cool! TIL. ✅